internal/mode/static/nginx/file/manager.go

```diff
@@ -88,7 +88,11 @@ func NewManagerImpl(logger logr.Logger, osFileManager OSFileManager) *ManagerImp
 func (m *ManagerImpl) ReplaceFiles(files []File) error {
 	for _, path := range m.lastWrittenPaths {
 		if err := m.osFileManager.Remove(path); err != nil {
-			return fmt.Errorf("failed to delete file %q: %w", path, err)
+			if os.IsNotExist(err) {
+				m.logger.V(1).Info("file not found, failed to delete file %q: %w", path, err)
```

Comment:
could you extend test for the manager so that they ensure no errors are returned when a file doesn't exist?